OpcUaClient/OpcUaInterface.hpp

```diff
@@ -218,14 +233,191 @@ class OpcUaWrapper : public OpcUaInterface {
     }
   }
 
-  void SubscriptionUnsubscribe(UA_Client *client, std::vector<int32_t> monItemIds, std::vector<int32_t> clientHandles) {
-    p_subscr->Unsubscribe(client, monItemIds, clientHandles);
-  }
-
-  UA_SessionState m_pSessionState;
-  UA_SecureChannelState m_pChannelState;
-};
-}  // namespace OpcUa
-}  // namespace Umati
+	void SubscriptionUnsubscribe(UA_Client *client, std::vector<int32_t> monItemIds, std::vector<int32_t> clientHandles){
+		p_subscr->Unsubscribe(client, monItemIds, clientHandles);
+	}
+
+	static void handler_events(UA_Client *client, UA_UInt32 subId, void *subContext, UA_UInt32 monId, void *monContext, size_t nEventFields, UA_Variant *eventFields) {
+		UA_NodeId* sourceNodeId;
+		UA_NodeId affectedNode;
+		UA_NodeId affectedType;
+		UA_Byte verb;
+		for(size_t i = 0; i < nEventFields; ++i) {
+			UA_Variant eventField = eventFields[i];
+			if(UA_Variant_hasScalarType(&eventField, &UA_TYPES[UA_TYPES_NODEID])) {
+				UA_NodeId* nodeId = (UA_NodeId *)eventField.data;
+				if(i == 1) {
+					sourceNodeId = nodeId;
+				}
+			}  else {
+				UA_ExtensionObject* extensionObjects = (UA_ExtensionObject*)eventField.data;
+				UA_ModelChangeStructureDataType* modelChangeStructureDataTypes = (UA_ModelChangeStructureDataType*)UA_Array_new(eventField.arrayLength, &UA_TYPES[UA_TYPES_MODELCHANGESTRUCTUREDATATYPE]);
+				for(int j = 0; j < eventField.arrayLength; j ++) {
+					UA_ExtensionObject extensionObject = extensionObjects[j];
+					UA_ModelChangeStructureDataType* modelChangeStructureDataType =(UA_ModelChangeStructureDataType*)extensionObject.content.decoded.data;
+					modelChangeStructureDataTypes[j] = *modelChangeStructureDataType;
+				}
+				OpcUaWrapper* pWrapper = static_cast<OpcUaWrapper*> (monContext);
+				if(pWrapper != nullptr && pWrapper->eventcallback != nullptr) {
+					for(int i = 0; i < eventField.arrayLength; i++) {
+						UA_ModelChangeStructureDataType modelChangeStructureDataType = modelChangeStructureDataTypes[i];
+						UA_NodeId affectedNode = modelChangeStructureDataType.affected;
+						UA_NodeId affectedType = modelChangeStructureDataType.affectedType;
+						UA_Byte verb = modelChangeStructureDataType.verb;
+						Umati::Dashboard::IDashboardDataClient::StructureChangeEvent stc;
+						std::map<uint16_t, std::string> id2Uri{{affectedNode.namespaceIndex, pWrapper->namespaceArray.at(affectedNode.namespaceIndex)}};
+						open62541Cpp::UA_NodeId affectedNodeCpp;
+						switch(affectedNode.identifierType) {
+							case UA_NODEIDTYPE_NUMERIC: affectedNodeCpp = open62541Cpp::UA_NodeId(affectedNode.namespaceIndex, affectedNode.identifier.numeric); break;
```

Comment:
Would any of these Methods do the same here?
https://github.com/umati/open62541Cpp/blob/5f3b7cf4a5fb7997036d40def96e6afbe8196e4a/Open62541Cpp/UA_NodeId.hpp#L27-L31